packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -359,12 +360,42 @@ export class OpenID4VCIClient {
     return response.successBody;
   }
 
+  /**
+   * Retrieve a supported credential by it's id. This is a convenience method that takes into account difference between the
+   * different versions of the oid4vci specification.
+   *
+   * NOTE: for v1_0-08, a single credential id in the issuer metadata could have multiple formats. When retrieving the
+   * supported credentials (using {@link getCredentialsSupported}), for v1_0-08, the format is appended to the id if
+   * there are multiple formats supported for that credential id. E.g. if the issuer metadata for v1_0-08 contains an
+   * entry with key `OpenBadgeCredential` and the supported formats are `jwt_vc-jsonld` and `ldp_vc`, then the id
+   * in the credentials supported will be `OpenBadgeCredential-jwt_vc-jsonld` and `OpenBadgeCredential-ldp_vc`, even though
+   * the offered credential is simply `OpenBadgeCredential`. This method takes this into account.
+   */
+  getCredentialsSupportedById(credentialSupportedId: string, format?: CredentialFormat): CredentialSupported[] {
+    return this.getCredentialMetadata(credentialSupportedId).filter((credentialSupported) => credentialSupported.format === format);
```

Comment:
What is the intent if the format is not supplied? I would expect it to return anything regardless of format